js/dashboard-controls.js

```diff
@@ -0,0 +1,223 @@
+/*
+ * Plantalytics
+ *     Copyright (c) 2016 Sapphire Becker, Katy Brimm, Scott Ewing,
+ *       Matt Fraser, Kelly Ledford, Michael Limb, Steven Ngo, Eric Turley.
+ *     This project is licensed under the MIT License.
+ *     Please see the file LICENSE in this distribution for license terms.
+ * Contact: plantalytics.capstone@gmail.com
+ */
+
+/////////////////////////////////// Controls ///////////////////////////////////
+
+$(function() {
+    $("#menu").click(function() {
+        var $menu = $(".menu");
+        var visible = $menu.is(":visible");
+        $menu.show();
+        var width = $menu.width();
+        var height = $menu.height();
+        if (!$menu.is(':animated')) {
+            if (visible) {
+                // Animate hiding menu.
+                $menu.animate({
+                    "height": 10,
+                }, {
+                    "duration": 150,
+                }).animate({
+                    "width": 0,
+                }, {
+                    "duration": 300,
+                    "queue": true,
+                    "complete": function() {
+                        $menu.width(width);
+                        $menu.height(height);
+                        $menu.hide();
+                        isAnimating = false;
+                    }
+                });
+            } else {
+                // Animate showing menu.
+                $menu.width(0).height(10);
+                $menu.animate({
+                    "width": width,
+                }, {
+                    "duration": 150,
+                }).animate({
+                    "height": height,
+                }, {
+                    "duration": 300,
+                    "queue": true
+                });
+            }
+        }
+    });
+
+    $(".data-view-defaults .data-button-label div").click(function(e) {
+        var parent = $(this).parent();
+        var selector = "." + parent.attr("class").replace(/ /g, ".");
+        if (localStorage.defaultDataView == selector) {
+            // Double tap to turn off.
+            delete localStorage.defaultDataView;
+            setTimeout(function() {
+                parent.find("input").prop("checked", false);
+            }, 10);
+        } else {
+            // Set new default.
+            localStorage.defaultDataView = selector;
+        }
+    });
+
+    $('#leafwetness-button').click(function() {
+        $.ajax({
```

Comment:
I feel like these buttons could be easily generalized into a single function. Rather than making the ajax call inline, something like getEnvData("leafwetness")